### PR TITLE
expression: implement vectorized evaluation for builtinNowWithArgSig

### PR DESCRIPTION
**Pull request by ekalinin**
PCP #12101

### What problem does this PR solve? 

This PR implements vectorized `builtinNowWithArgSig`

### What is changed and how it works?

```sh
➜ make vectorized-bench VB_FILE=Time VB_FUNC=builtinNowWithArgSig
cd ./expression && \
	go test -v -benchmem \
		-bench=BenchmarkVectorizedBuiltinTimeFunc \
		-run=BenchmarkVectorizedBuiltinTimeFunc \
		-args "builtinNowWithArgSig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-12    	1000000000	         0.00699 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinNowWithArgSig-VecBuiltinFunc-12         	    3584	    323626 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinNowWithArgSig-NonVecBuiltinFunc-12      	    3280	    367534 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	3.420s
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -563,11 +563,44 @@ func (b *builtinTimeTimeTimeDiffSig) vecEvalDuration(input *chunk.Chunk, result
...
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufFsp)
+	if err = b.args[0].VecEvalInt(b.ctx, input, bufFsp); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(bufFsp)
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: I think this line is wrong. If the fsp is null, set the fsp to 0 ,but not set the result to null.

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -563,11 +563,44 @@ func (b *builtinTimeTimeTimeDiffSig) vecEvalDuration(input *chunk.Chunk, result
...
+	times := result.Times()
+	fsps := bufFsp.Int64s()
+
+	for i := 0; i < n; i++ {
+		fsp := int8(0)
+		if !result.IsNull(i) {
+			fsp = int8(fsps[i])
+		}
+
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: Need check `fsp > int64(types.MaxFsp)` or `fsp < int64(types.MinFsp)`

File: expression/builtin_time_vec.go [hunk 3]
```diff
@@ -563,11 +563,44 @@ func (b *builtinTimeTimeTimeDiffSig) vecEvalDuration(input *chunk.Chunk, result
...
+		t, isNull, err := evalNowWithFsp(b.ctx, fsp)
+		if isNull {
+			result.SetNull(i, true)
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: Dealing with the err should be first, and then check isNull.

File: expression/builtin_time_vec.go [hunk 4]
```diff
@@ -563,11 +563,44 @@ func (b *builtinTimeTimeTimeDiffSig) vecEvalDuration(input *chunk.Chunk, result
...
+	result.ResizeTime(n, false)
+	result.MergeNulls(bufFsp)
+	times := result.Times()
+	fsps := bufFsp.Int64s()
+
+	for i := 0; i < n; i++ {
+		fsp := int8(0)
+		if !result.IsNull(i) {
+			fsp = int8(fsps[i])
+		}
```
Comment (b41sh) [inline expression/builtin_time_vec.go]: we need to check whether fsp is in valid range

**Comment by qw4990** [comment]
Hi, @ekalinin , we noticed that recently you are very active in our community and have contributed 15 PRs for the expression package!!! Thanks a lot for your contribution and we'd like to invite you to join our [`sig-expr Slack channel`](https://tidbcommunity.slack.com/archives/CMRD79DRR) so that if you have any questions related to expression, you can discuss with us and other contributors. Have a nice day!

File: expression/builtin_time_vec.go (hunk 4 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Yeah, missed it. Fixed. Thanks.

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Got it. Delete it. Used directly `bufFsp` to check if current fsp is null or not.

File: expression/builtin_time_vec.go (hunk 2 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Fixed. Thanks.

File: expression/builtin_time_vec.go (hunk 3 above)
Comment (ekalinin) [reply expression/builtin_time_vec.go]: Sure. Fixed.

**Comment by b41sh** [comment]
LGTM

**Pull request merged**